Unable to scaffold chain on mac

**Describe the bug**
When trying to scaffold chain on mac I see that it is trying to scaffold chain but then after a while I get:
```
...
TypeError: require(...).internalModuleReadJSON is not a function
...
--ts_proto_out: protoc-gen-ts_proto: Plugin failed with status code 1.
: exit status 1
```
(full output below)

**To Reproduce**
Steps to reproduce the behavior:
1. Use a mac with OSX 12
2. run `ignite scaffold chain github.com/alice/checkers`

**Please provide the version output**
 - `ignite version`: tried with 17, 20, and 21 - always happens
 - `go version`: go version go1.18.2 darwin/amd64
 
 **What have you tried**
uninstalling & reinstalling go via homebrew
removing all go modules from `~/go`
using a bunch of different node versions
uninstalling & reinstalling ignite

full output of `ignite scaffold chain github.com/alice/checkers`:
```
error while running command /var/folders/20/4dg_fzn57274yrptbr8wv93m0000gn/T/protoc2752117165 -I /var/folders/20/4dg_fzn57274yrptbr8wv93m0000gn/T/263921578 --plugin /var/folders/20/4dg_fzn57274yrptbr8wv93m0000gn/T/protoc-gen-ts_proto -I /Users/nikitajerschow/Documents/PassiveIncome/SendblueBase/sendblue-blockchain/moonlink/proto -I /Users/nikitajerschow/go/pkg/mod/github.com/cosmos/cosmos-sdk@v0.45.4/proto -I /Users/nikitajerschow/go/pkg/mod/github.com/cosmos/cosmos-sdk@v0.45.4/third_party/proto --ts_proto_out=. /Users/nikitajerschow/Documents/PassiveIncome/SendblueBase/sendblue-blockchain/moonlink/proto/moonlink/genesis.proto /Users/nikitajerschow/Documents/PassiveIncome/SendblueBase/sendblue-blockchain/moonlink/proto/moonlink/params.proto /Users/nikitajerschow/Documents/PassiveIncome/SendblueBase/sendblue-blockchain/moonlink/proto/moonlink/query.proto /Users/nikitajerschow/Documents/PassiveIncome/SendblueBase/sendblue-blockchain/moonlink/proto/moonlink/tx.proto --ts_proto_opt=snakeToCamel=false: moonlink/query.proto:6:1: warning: Import cosmos/base/query/v1beta1/pagination.proto is unused.
node:internal/modules/package_json_reader:4
const internalModuleReadJSON = function (f) { return require('fs').internalModuleReadJSON(f); };
                                                                   ^

TypeError: require(...).internalModuleReadJSON is not a function
    at internalModuleReadJSON (node:internal/modules/package_json_reader:4:68)
    at Object.read (node:internal/modules/package_json_reader:21:42)
    at readPackage (node:internal/modules/cjs/loader:296:36)
    at readPackageScope (node:internal/modules/cjs/loader:329:19)
    at trySelf (node:internal/modules/cjs/loader:444:40)
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:910:24)
    at Function.Module._load (node:internal/modules/cjs/loader:778:27)
    at Module.require (node:internal/modules/cjs/loader:1005:19)
    at Module._preloadModules (node:internal/modules/cjs/loader:1276:12)
    at loadPreloadModules (node:internal/bootstrap/pre_execution:489:5)
--ts_proto_out: protoc-gen-ts_proto: Plugin failed with status code 1.
: exit status 1
```

--- a/ignite/pkg/cosmosgen/generate_javascript.go
+++ b/ignite/pkg/cosmosgen/generate_javascript.go
@@ -126,6 +126,7 @@ func (g *jsGenerator) generateModule(ctx context.Context, tsprotoPluginPath, app
 		includePaths,
 		tsOut,
 		protoc.Plugin(tsprotoPluginPath, "--ts_proto_opt=snakeToCamel=false"),
+		protoc.Env("NODE_OPTIONS="), // unset nodejs options to avoid unexpected issues with vercel "pkg"
 	)
 	if err != nil {
 		return err

--- a/ignite/pkg/protoc/protoc.go
+++ b/ignite/pkg/protoc/protoc.go
@@ -23,6 +23,7 @@ type configs struct {
 	pluginPath             string
 	isGeneratedDepsEnabled bool
 	pluginOptions          []string
+	env                    []string
 }
 
 // Plugin configures a plugin for code generation.
@@ -38,6 +39,13 @@ func Plugin(path string, options ...string) Option {
 func GenerateDependencies() Option {
 	return func(c *configs) {
 		c.isGeneratedDepsEnabled = true
+	}
+}
+
+// Env assigns environment values during the code generation.
+func Env(v ...string) Option {
+	return func(c *configs) {
+		c.env = v
 	}
 }
 
@@ -119,10 +127,15 @@ func Generate(ctx context.Context, outDir, protoPath string, includePaths, proto
 		command = append(command, files...)
 		command = append(command, c.pluginOptions...)
 
-		if err := exec.Exec(ctx, command,
+		execOpts := []exec.Option{
 			exec.StepOption(step.Workdir(outDir)),
 			exec.IncludeStdLogsToError(),
-		); err != nil {
+		}
+		if c.env != nil {
+			execOpts = append(execOpts, exec.StepOption(step.Env(c.env...)))
+		}
+
+		if err := exec.Exec(ctx, command, execOpts...); err != nil {
 			return err
 		}
 	}

